rust/opendp/src/meas/mod.rs

```diff
@@ -0,0 +1,122 @@
+        };
+        let input_metric = L1Sensitivity::new();
+        let output_measure = MaxDivergence::new();
+        let privacy_relation = move |d_in: &f64, d_out: &f64| *d_out >= *d_in / sigma;
+        Measurement::new(input_domain, output_domain, function, input_metric, output_measure, privacy_relation)
+    }
+}
+
+pub struct VectorLaplaceMechanism<T> {
+    data: PhantomData<T>
+}
+
+// laplace for vector-valued query
+impl<T> MakeMeasurement1<VectorDomain<AllDomain<T>>, VectorDomain<AllDomain<T>>, L1Sensitivity<f64>, MaxDivergence, f64> for VectorLaplaceMechanism<T>
+    where T: Copy + NumCast {
+    fn construct(sigma: f64) -> Measurement<VectorDomain<AllDomain<T>>, VectorDomain<AllDomain<T>>, L1Sensitivity<f64>, MaxDivergence> {
+        let input_domain = VectorDomain::new_all();
+        let output_domain = VectorDomain::new_all();
+        let function = move |arg: &Vec<T>| -> Vec<T> {
+            arg.iter()
+                .map(|v| <f64 as NumCast>::from(*v).and_then(|v| T::from(v + laplace(sigma))))
+                .collect::<Option<_>>().unwrap()
+        };
+        let input_metric = L1Sensitivity::new();
+        let output_measure = MaxDivergence::new();
+        let privacy_relation = move |d_in: &f64, d_out: &f64| *d_out >= *d_in / sigma;
+        Measurement::new(input_domain, output_domain, function, input_metric, output_measure, privacy_relation)
+    }
+}
+
+pub struct GaussianMechanism<T> {
+    data: PhantomData<T>
+}
+
+// gaussian for scalar-valued query
+impl<T> MakeMeasurement1<AllDomain<T>, AllDomain<T>, L2Sensitivity<f64>, SmoothedMaxDivergence, f64> for GaussianMechanism<T>
+    where T: Copy + NumCast {
+    fn construct(sigma: f64) -> Measurement<AllDomain<T>, AllDomain<T>, L2Sensitivity<f64>, SmoothedMaxDivergence> {
+        let input_domain = AllDomain::new();
+        let output_domain = AllDomain::new();
+        let function = move |arg: &T| -> T {
+            // TODO: switch to gaussian
+            <f64 as NumCast>::from(*arg).and_then(|v| T::from(v + laplace(sigma))).unwrap()
+        };
+        let input_metric = L2Sensitivity::new();
+        let output_measure = SmoothedMaxDivergence::new();
+        // https://docs.google.com/spreadsheets/d/132rAzbSDVCKqFZWeE-P8oOl9f23PzkvNwsrDV5LPkw4/edit#gid=0
+        let privacy_relation = move |d_in: &f64, d_out: &(f64, f64)| {
+            let (eps, delta) = d_out.clone();
+            eps.min(1.) >= (*d_in / sigma) * (2. * (1.25 / delta).ln()).sqrt()
+        };
+        Measurement::new(input_domain, output_domain, function, input_metric, output_measure, privacy_relation)
+    }
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_make_base_laplace() {
+        let measurement = LaplaceMechanism::<f64>::construct(1.0);
+        let arg = 0.0;
+        let _ret = measurement.function.eval(&arg);
+        // TODO: Test for base_laplace
+    }
+
```

Comment:
Can we add a test for VectorLaplaceMechanism too? (I know it's just a sanity check, but it's a place to put a TODO anyway.)

Also would be good to do one for GaussianMechanism.